nautobot_floor_plan/static/nautobot_floor_plan/js/floorplan.js

```diff
@@ -290,134 +304,426 @@ document.addEventListener('DOMContentLoaded', function() {
+            if (anchorElement) {
+                // Ensure scroll happens after zoom animation
+                gsap.delayedCall(1.1, () => {  // Delay to ensure zoom is complete
+                    anchorElement.scrollIntoView({ behavior: "smooth", block: "start" });
+                });
+            }
+        }
+    }
+    // Disable zoom and pan interactions
+    function disableZoomAndPan() {
+        // Temporarily remove mouse listeners for zoom and pan
+        document.querySelector("#floor-plan-svg").style.pointerEvents = "none"; // Disable pointer events
+    }
+
+    // Enable zoom and pan interactions
+    function enableZoomAndPan() {
+        // Restore mouse listeners for zoom and pan
+        document.querySelector("#floor-plan-svg").style.pointerEvents = "auto"; // Re-enable pointer events
+    }
+    // Function for complete cleanup of all highlighting effects
+    function completeCleanup(element, effectElements, animations) {
+        console.log("Performing complete cleanup of highlighting");
+    
+        // Stop all animations
+        animations.forEach(anim => {
+            if (anim && typeof anim.cancel === 'function') {
+                anim.cancel();
+            }
+        });
+    
+        // Remove all visual effect elements
+        effectElements.forEach(el => {
+            if (el && el.parentNode) {
+                el.parentNode.removeChild(el);
+            }
+        });
+    
+        // If element exists, restore its original styling
+        if (element) {
+            element.classList.remove('highlighted');
+            element.classList.remove('static-highlight');
+            element.removeAttribute('data-highlighted');
+        }
+    
+        console.log("Cleanup complete");
+    }
+
+    // Create all visual highlight effects and return references to animations and elements
+    function createHighlightEffects(element, svg) {
+        const effectElements = [];
+        const effectAnimations = [];
+
+        try {
+            // Get element's bounding box
+            const bbox = element.getBBox();
+            const centerX = bbox.x + bbox.width / 2;
+            const centerY = bbox.y + bbox.height / 2;
+
+            // Create a spotlight circle
+            const spotlight = document.createElementNS("http://www.w3.org/2000/svg", "circle");
+            spotlight.setAttribute("cx", centerX);
+            spotlight.setAttribute("cy", centerY);
+            spotlight.setAttribute("r", Math.max(bbox.width, bbox.height) * 1.5);
+            spotlight.setAttribute("fill", "rgba(255, 255, 0, 0.15)");
+            spotlight.setAttribute("class", "spotlight-effect");
+            spotlight.setAttribute("pointer-events", "none");
+            spotlight.setAttribute("data-highlight-effect", "true");
+
+            // Insert spotlight at the beginning of SVG
+            if (svg.firstChild) {
+                svg.insertBefore(spotlight, svg.firstChild);
```

Comment:
```suggestion
            spotlight.setAttribute("fill", "rgba(255, 255, 0, 0.8)");
```

How would you feel about starting with a higher opacity value? the Just thinking of making the spotlight darker in general. I love the effect by the way :heart: